src/test/java/emissary/client/HTTPConnectionFactoryTest.java

```diff
@@ -90,6 +90,21 @@ void loadPemCertFile() throws Exception {
         assertNotSame(SSLContext.getDefault(), fromConfig);
     }
 
+    @Test
+    void loadPemMultiCertsAndCommentsFile() throws Exception {
+        addKeystoreProps(this.cfg);
+        addTrustStoreProps(this.cfg);
+
+        this.cfg.removeEntry(CFG_TRUST_STORE, "*");
+        this.cfg.addEntry(CFG_TRUST_STORE, projectBase + "/test-classes/certs/testcertwithcomments.pem");
+
+        final HTTPConnectionFactory instance = new HTTPConnectionFactory(this.cfg);
+
+        final SSLContext fromConfig = instance.build(this.cfg);
+
+        assertNotSame(SSLContext.getDefault(), fromConfig);
```

Comment:
I'll admit, it wouldn't be awful if we could more directly validate the certificate loading, but that would require the transient TrustStore to be accessible or require us to refactor the KeyStore management into a separate utility class that we could test separately.

Reply:
The long term plan is to move this the KeyStore methods to a utility class. I have no issue with including that in this merge request.

Reply:
Moved PKI functions to a util class and created additional unit-tests.